wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -1481,4 +1490,103 @@ private void closeStream(InputStream mnlistinfoBootStrapStream) {
             }
         }
     }
+
+    // TODO: should we have a backup blockchain file?
+//    private NewBestBlockListener newBestBlockListener = block -> {
+//        try {
+//            backupBlockStore.put(block);
+//        } catch (BlockStoreException x) {
+//            throw new RuntimeException(x);
+//        }
+//    };
+
+    private boolean verifyBlockStore(BlockStore store) throws BlockStoreException {
+            StoredBlock cursor = store.getChainHead();
+            for (int i = 0; i < 10; ++i) {
+                cursor = cursor.getPrev(store);
+                if (cursor == null || cursor.getHeader().equals(Constants.NETWORK_PARAMETERS.getGenesisBlock())) {
+                    break;
+                }
+            }
+            return true;
+    }
```

Comment:
This was added to check to see if it was possible to access the blockStore.  The theory was that the blockStore file was corrupted and this resulted in a freeze when accessing the file.

Reply:
So far this test passed, even with a device with sync problems.